### ISSUE TITLE
No C# project is currently loaded when .NET 8 Preview 7 is present

### ISSUE THREAD

**Issue by timheuer** [microsoft/vscode-dotnettools#318]
From roslyn created by [richlander](https://github.com/richlander): dotnet/roslyn#69379  

Got this.

<img width="465" alt="image" src="https://github.com/dotnet/roslyn/assets/2608468/ad135de6-00f5-4c79-ac30-6c7cb2dbc93c">

Takes me to this fun page: https://www.bing.com/?ref=aka&shorturl=vscode/dotnet/debug

I wasn't able to self-diagnose my problem from the information there.

I am seeing this with multiple projects.

Also can exhibit as this in Solution Explorer view:
![image](https://github.com/microsoft/vscode-dotnettools/assets/4821/ab9c9f35-bccb-4f1f-9d5b-c66fa73cd3be)

```
Version: 1.81.0 (Universal)
Commit: 6445d93c81ebe42c4cbd7a60712e0b17d9463e97
Date: 2023-08-02T12:40:02.782Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Darwin arm64 22.6.0
```

Dev Kit: v0.4.2

**Comment by richlander** [comment]
I created a new project with `dotnet new console` and this started happening. I then went back to another .NET 8 project that I had been using and saw the same thing. I updated macOS to latest today (13.5) and I believe VS Code updated today as well. I did not update .NET today.

**Comment by richlander** [comment]
I played with this a bit more.

I went back to Preview 6. Still repros. Went back to the non pre-release DevKit. Problem goes away. It appears that this issue isn't related to Preview 7.

**Comment by lambdageek** [comment]
Is there a workaround?  Disabling `ms-dotnettools.dotnet-maui` and `ms-dotnettools.csdevkit`, but leaving `ms-dotnettools.csharp` enabled doesn't help.

**Comment by chrisntr** [comment]
I'm seeing what might be a related and/or similar problem, unfortunately I can't copy and paste the error message but here's a screenshot of it: 
<img width="1051" alt="image" src="https://github.com/microsoft/vscode-dotnettools/assets/18434/0b6d614c-39b1-4ac4-9e2d-ceb3cabf2aea">

I did just install Preview 7 too: dotnet --version    
8.0.100-preview.7.23376.3

**Comment by timheuer** [comment]
Yes @chrisntr same issue.

**Comment by aozametaneko** [comment]
I got a similar message too. I solved it by adding the following files, launch.json and Task.json.
It is a Japanese site, but please refer to it.
https://blog.ytabuchi.dev/entry/2017/02/24/150000

Please change the dll path in the following part of launch.json to the path of your own file.
"program": "${workspaceRoot}\\bin\\Debug\\netcoreapp1.1\\NetCoreSample.dll",

**Comment by MeydanOzeri** [comment]
Any workaround until fixed ?

**Comment by mattleibow** [comment]
This is probably related to https://github.com/dotnet/runtime/pull/87857

That was a breaking change between p6 and p7 because they move/required a new attribute or something. But basically, all dlls compiled with p6 need to be recompiled with p7 - I think?

**Comment by alexrp** [comment]
@timheuer possibly silly question, but what is the new setting called?

**Comment by AndiLilaj** [comment]
> @timheuer possibly silly question, but what is the new setting called?

Just changed it myself and already forgot 😂
Along the lines of Specify SDK version

**Comment by alexrp** [comment]
Found it: `dotnet.preferRuntimeFromSDK`

Works for me.

**Comment by timheuer** [comment]
![image](https://github.com/microsoft/vscode-dotnettools/assets/4821/0e02b4fe-67d2-411d-87f0-7d85b913f272)

**Comment by alexbocharov** [comment]
@timheuer It's works

**Comment by Michael-Eng** [comment]
@lifengl can you update this bug?

**Comment by Michael-Eng** [comment]
Closing this as fixed with prerelease version 0.4.5

**Comment by maiquitome** [comment]
I removed `.net` from `asdf` and installed `.net` directly from the `microsoft website` and it worked for me. 🎉
